src/discussions/posts/post/Post.jsx

```diff
@@ -86,31 +86,54 @@ function PostHeader({
           </span>
         </div>
       </div>
-      <div className="d-flex mr-3">
-        <Icon src={QuestionAnswer} />
-        <span style={{ minWidth: '2rem' }}>
-          {post.commentCount}
-        </span>
-      </div>
+      {preview
+        ? (
+          <div className="d-flex">
+            <Icon src={QuestionAnswer} />
+            <span style={{ minWidth: '2rem' }}>
+              {post.commentCount}
+            </span>
+          </div>
+        )
+        : <ActionsDropdown commentOrPost={post} actionHandlers={actionHandlers} />}
     </div>
   );
 }
 
 PostHeader.propTypes = {
   intl: intlShape.isRequired,
   post: postShape.isRequired,
+  preview: PropTypes.bool.isRequired,
+  actionHandlers: PropTypes.objectOf(PropTypes.func).isRequired,
 };
 
 function Post({
   post,
+  preview,
   intl,
 }) {
+  const location = useLocation();
+  const history = useHistory();
   const dispatch = useDispatch();
+  const actionHandlers = {
+    [EditActions.EDIT_CONTENT]: () => history.push(`${location.pathname}/edit`),
+    [EditActions.DELETE]: () => dispatch(removeThread(post.id)),
+    [EditActions.CLOSE]: () => dispatch(updateExistingThread(post.id, { closed: !post.closed })),
+    [EditActions.PIN]: () => dispatch(updateExistingThread(post.id, { pinned: !post.pinned })),
+    [EditActions.REPORT]: () => dispatch(updateExistingThread(post.id, { flagged: !post.abuseFlagged })),
+  };
 
   return (
     <div className="d-flex flex-column p-2.5 w-100">
-      <PostHeader post={post} intl={intl} />
-      <div className="mt-2 mb-0 p-0" dangerouslySetInnerHTML={{ __html: post.renderedBody }} />
+      <PostHeader post={post} intl={intl} preview={preview} actionHandlers={actionHandlers} />
+      <div
+        className="d-block mt-2 mb-0 p-0 overflow-hidden text-break"
+        dangerouslySetInnerHTML={{ __html: post.renderedBody }}
+        style={{
+          maxHeight: preview ? '3rem' : null,
+          maxWidth: preview ? '80%' : null,
+        }}
+      />
```

Comment:
This is somewhat temporary, we should instead generate a preview on the server side.